src/main/java/de/maxhenkel/peek/data/DataStore.java

```diff
@@ -1,12 +1,14 @@
 package de.maxhenkel.peek.data;
 
+import net.minecraft.core.NonNullList;
+import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.level.block.entity.ShulkerBoxBlockEntity;
 
 import javax.annotation.Nullable;
 
 public class DataStore {
-
     @Nullable
     public static ShulkerBoxBlockEntity lastOpenedShulkerBox;
-
+    public static NonNullList<ItemStack> ecinv = NonNullList.withSize(27, ItemStack.EMPTY);
+    public static boolean enderChestOpen = false;
```

Comment:
I don't know if this is the best solution. There are cases where this can be true without having an ender chest open.

Reply:
You're right, I clearly didn't check thoroughly enough. The value is de synced when the player moves far enough from the ender chest for it to close automatically.

I really appreciate this note! I realize that I can use the same approach I used in `onHandleContainerContent` in `onHandleInventoryMouseClick` to detect if an ender chest is opened. This eliminates the need to independently track if an ender chest is open. The only way I can see this going wrong is if a container screen has the same translation key as an ender chest. In my opinion, this is acceptable behavior as the client is being told the container screen is an ender chest.